slip10/utils.py

```diff
@@ -92,43 +109,136 @@ def derive_public_child(self, pubkey, chaincode, index):
         """
-        from ecdsa.ellipticcurve import INFINITY
-
         assert isinstance(pubkey, bytes) and isinstance(chaincode, bytes)
         if index & HARDENED_INDEX != 0:
             raise SLIP10DerivationError("Hardened derivation is not possible.")
 
-        # payload is the I from the SLIP. Index is 32 bits unsigned int, BE.
         payload = hmac.new(
             chaincode, pubkey + index.to_bytes(4, "big"), hashlib.sha512
         ).digest()
+        base_point = self._bytes_to_point(pubkey)
         while True:
             tweak = int.from_bytes(payload[:32], "big")
-            point = ecdsa.VerifyingKey.from_string(pubkey, self.curve).pubkey.point
-            point += self.curve.generator * tweak
-            if tweak <= self.curve.order and point != INFINITY:
+            tweak_point = self._scalar_mult(tweak, self.generator)
+            child_point = self._point_add(base_point, tweak_point)
+            if tweak <= self.order and child_point is not None:
                 break
             payload = hmac.new(
                 chaincode,
                 b"\x01" + payload[32:] + index.to_bytes(4, "big"),
                 hashlib.sha512,
             ).digest()
-        return point.to_bytes("compressed"), payload[32:]
+        return self._point_to_bytes(child_point), payload[32:]
 
     def privkey_is_valid(self, privkey):
         key = int.from_bytes(privkey, "big")
-        return 0 < key < self.curve.order
+        return 0 < key < self.order
 
     def pubkey_is_valid(self, pubkey):
         try:
-            ecdsa.VerifyingKey.from_string(pubkey, self.curve)
-            return True
-        except ecdsa.errors.MalformedPointError:
+            point = self._bytes_to_point(pubkey)
+        except ValueError:
             return False
+        return point is not None and self._is_on_curve(point)
 
     def privkey_to_pubkey(self, privkey):
-        sk = ecdsa.SigningKey.from_string(privkey, self.curve)
-        return sk.get_verifying_key().to_string("compressed")
+        if not self.privkey_is_valid(privkey):
+            raise ValueError("Invalid private key")
+        scalar = int.from_bytes(privkey, "big")
+        point = self._scalar_mult(scalar, self.generator)
+        if point is None:
+            raise ValueError("Point at infinity")
+        return self._point_to_bytes(point)
+
+    def _is_on_curve(self, point: Point) -> bool:
+        if point is None:
+            return True
+        x, y = point
+        return (y * y - (x * x * x + self.a * x + self.b)) % self.p == 0
+
+    def _point_add(self, p1: Point, p2: Point) -> Point:
+        if p1 is None:
+            return p2
+        if p2 is None:
+            return p1
+        if p1 == p2:
+            return self._point_double(p1)
+
+        x1, y1 = p1
+        x2, y2 = p2
+        if x1 == x2:
+            return None
+
+        slope = ((y2 - y1) * self._inverse_mod((x2 - x1) % self.p)) % self.p
```

Comment:
`%` is unnecessary since it happens in `_inverse_mod()`.
```suggestion
        slope = ((y2 - y1) * self._inverse_mod(x2 - x1)) % self.p
```